### DISCUSSION TITLE
Better/More example for starter

### DISCUSSION THREAD

**Discussion by truonghoangduy** [PhilipsHue/flutter_reactive_ble#416] [Ideas]
Looking in the two example in the flutter_reactive_ble (in the repo, and the unofficial) it too complicated. Yes flutter_reactive_ble/example contained a **deep look** in the API and the way of the library build up and how it could handle scanning, establish connection, pub/sub to a characteristic.

The idea is open for community contribute example influence by [bloc](https://github.com/felangel/bloc/tree/master/examples). where they hold amazing example for starter to learn about the API and how to handle them.

What i **could contribute is an example** using flutter_reactive_ble just to **scan BLE devices which advertise the Battery UUID** 🔋.

Such a small demo but i think it would help other learn a lot form the API. I could list things that starter may interest 🛎️. 

- Using the library Dev must has to use some production BLE devices or may have them around their house such as (Apple watch, Samsung watch,....). Example run it will pop devices around their house.
- Why the API scanning required UUID -> common services like battery will pop up and what about establish connection via device name/id.
- Read characteristic.
- Simpler Widget design/API interaction so that the **codebase easier to read**.
- Design an example such beginner could easier too attach debug and figure it out how the API was interacted.
- Better example readme explaining the stuff of it doing.

Seen _Bluetooth in common is hard and quite niche_ and with Flutter it gonna be harder. I wonder does my idea of proposal is considerable just let me know 💭.

**Comment by remonh87** [comment]
@truonghoangduy  thanks for your proposal. 

The reason why I would like to keep the example app with that much content is: 

- It gives a complete overview of the API and provides a clean architecture pattern with good separation of concerns. Of course things can be improved but we use a very basic state management solution (Provider) in order to prevent users to need to learn another library.
- We need an app that helps reproducing issues. BLE is quite complex on itself and it is easy to run into issues. The example app provides a common base where we can determine whether it is a bug or a usage issue

I think the Current app should be improved see #404 . 

The reason why we do not want to go for a BLOC approach is that we simply lack the time to maintain all these examples in our repo (creating is one thing but keeping them up to date is something different). Feel free to create such an app on your own GitHub and we are happy to share it in the readme (as we did with the other example as well) . I think it would be awesome to have more examples and if you need help feel free to reach out.

**Reply by farr64** [reply]
I concur with @remonh87.

Please see the comment I just posted on #404
